docs/hello_nextflow/09_hello_nf-core.md

```diff
@@ -244,4 +244,1050 @@ You have a general idea of what nf-core offers and you know how to run an nf-cor
+    │   ├── meta.yml
+    │   └── tests
+    │       ├── main.nf.test
+    │       ├── main.nf.test.snap
+    │       └── tags.yml
+    .
+```
+
+Modules from nf-core follow a similar same structure and contain a small number of additional files that are used for testing using [nf-test](https://www.nf-test.com/) and documentation about the module.
+
+!!!note
+
+    Some nf-core modules are also split into command specific directories:
+
+    ```console
+    │
+    └── <tool name>
+        └── <command>
+            ├── environment.yml
+            ├── main.nf
+            ├── meta.yml
+            └── tests
+                ├── main.nf.test
+                ├── main.nf.test.snap
+                └── tags.yml
+    ```
+
+!!!note
+
+    The nf-core template does not come with a local modules folder by default.
+
+#### Configuration files
+
+The nf-core pipeline template utilizes Nextflow's flexible customization options and has a series of configuration files throughout the template.
+
+In the template, the `nextflow.config` file is a central configuration file and is used to set default values for parameters and other configuration options. The majority of these configuration options are applied by default while others (e.g., software dependency profiles) are included as optional profiles.
+
+There are several configuration files that are stored in the `conf` folder and are added to the configuration by default or optionally as profiles:
+
+-   `base.config`: A 'blank slate' config file, appropriate for general use on most high performance compute environments.
+-   `modules.config`: Additional module directives and arguments.
+-   `test.config`: A profile to run the pipeline with minimal test data.
+-   `test_full.config`: A profile to run the pipeline with a full-sized test dataset.
+
+#### `nextflow_schema.json`
+
+The `nextflow_schema.json` is a file used to store parameter related information including type, description and help text in a machine readable format. The schema is used for various purposes, including automated parameter validation, help text generation, and interactive parameter form rendering in UI interfaces.
+
+### Takeaway
+
+You have now created a template pipeline, and learned about important template files
+
+### What's next?
+
+Congratulations! You have now created a template pipeline. In the next step, we will start adding new tools to it. 
+
+---
+
+## Check the input data
+
+Above, we said that the `test` profile comes with small test files that are stored in the nf-core. Let's check what type of files we are dealing with to plan our expansion. We can inspect any channel content by using the `dump` operator: 
+
+```groovy title="workflows/myfirstpipeline.nf" linenums="27"
+ch_samplesheet.dump()
+```
+
+We can then run the pipeline with the additional argument `-dump-channels`. This will run `dump()`. It is great for development, because a normal user will not add this flag and thus not see the channel content:
```

Comment:
```suggestion
We can then run the pipeline with the additional argument `-dump-channels`. This will run `dump()`. It is great for development, because a normal user will not add this flag and thus not see the channel content:
```

Think I saw you ask about this somewhere- feels like it lives somewhere not nf-core specific

Reply:
yep @kenibrewer do you time to add the dump-channels to hello-nextflow?

Reply:
Added to hello-nextflow in #450 and just keeping it short here. Main purpose is to check what sort of data we get

Reply:
reverted to `view` as discussed on slack